src/character.cpp

```diff
@@ -9263,75 +9297,90 @@ int Character::run_cost( int base_cost, bool diag ) const
 
     if( !is_mounted() ) {
         if( movecost > 105 ) {
-            movecost *= mutation_value( "movecost_obstacle_modifier" );
+            run_cost_effect_mul( movecost, effects, mutation_value( "movecost_obstacle_modifier" ),
+                                 _( "Obstacle Mutations" ) );
 
             if( has_proficiency( proficiency_prof_parkour ) ) {
-                movecost *= .5;
+                run_cost_effect_mul( movecost, effects, 0.5, _( "Parkour" ) );
             }
 
             if( movecost < 100 ) {
+                run_cost_effect effect { _( "Bonuses Capped" ) };
+                effects.push_back( effect );
```

Comment:
The three run_cost_effect entries above here are actually never used, because draw_speed_tab always calls this function with a base movecost of 100, but I wanted to be consistent with the later cases.